pkg/services/sqlstore/search_builder.go

```diff
@@ -88,15 +90,28 @@ func (sb *SearchBuilder) ToSql() (string, []interface{}) {
 		sb.buildMainQuery()
 	}
 
+	sb.sql.WriteString(`
+		ORDER BY dashboard.id ` + sb.getLimitExpression() + `) as ids
+		INNER JOIN dashboard on ids.id = dashboard.id
+	`)
+
 	sb.sql.WriteString(`
 		LEFT OUTER JOIN dashboard folder on folder.id = dashboard.folder_id
 		LEFT OUTER JOIN dashboard_tag on dashboard.id = dashboard_tag.dashboard_id`)
 
-	sb.sql.WriteString(" ORDER BY dashboard.title ASC" + dialect.Limit(5000))
```

Comment:
yes, this is a small breaking change, calls with a limit > 5000 will now return an error when before they did not as this limit was kind of silent. 

It's actually a bit complicated now that I think about it. The limit the user could specify had no upper bound, but we had this outer hardcoded limit (5000) on the outer query that is actually on the wrong query. The outer query does a outer join with tags, so depending on how many tags the dashboards have this 5000 limit will act very differently. Technically if you have 10 tags on each dashboard it would limit to only 500 dashboards. That is why this limit is now removed.

Reply:
yea, this was the strange outer limit that should not be here as it depends on number of tags